plugin/src/main/java/io/jenkins/plugins/casc/impl/secrets/VaultSecretSource.java

```diff
@@ -39,114 +43,116 @@
 
         // Parse variables
-        Optional<String> vaultPw = getVariable(CASC_VAULT_PW, prop);
-        Optional<String> vaultUser = getVariable(CASC_VAULT_USER, prop);
+        Optional<String> vaultEngineVersionOpt = getVariable(CASC_VAULT_ENGINE_VERSION, prop);
         Optional<String> vaultUrl = getVariable(CASC_VAULT_URL, prop);
-        Optional<String> vaultMount = getVariable(CASC_VAULT_MOUNT, prop);
-        Optional<String> vaultToken = getVariable(CASC_VAULT_TOKEN, prop);
-        Optional<String> vaultAppRole = getVariable(CASC_VAULT_APPROLE, prop);
-        Optional<String> vaultAppRoleSecret = getVariable(CASC_VAULT_APPROLE_SECRET, prop);
         Optional<String> vaultNamespace = getVariable(CASC_VAULT_NAMESPACE, prop);
-        Optional<String> vaultEngineVersion = getVariable(CASC_VAULT_ENGINE_VERSION, prop);
-
         Optional<String[]> vaultPaths = getCommaSeparatedVariables(CASC_VAULT_PATHS, prop)
                 .map(Optional::of)
                 .orElse(getCommaSeparatedVariables(CASC_VAULT_PATH, prop)); // TODO: deprecate!
 
         // Check mandatory variables are set
         if (!vaultUrl.isPresent() || !vaultPaths.isPresent()) return;
 
-        // Check defaults
-        if (!vaultMount.isPresent()) vaultMount = Optional.of(DEFAULT_USER_BACKEND);
-        if (!vaultEngineVersion.isPresent()) vaultEngineVersion = Optional.of(DEFAULT_ENGINE_VERSION);
+        String vaultEngineVersion = vaultEngineVersionOpt.orElse(DEFAULT_ENGINE_VERSION);
+        this.vaultPaths = vaultPaths.get();
+        determineAuthenticator(prop);
 
         // configure vault client
-        VaultConfig vaultConfig = new VaultConfig().address(vaultUrl.get());
+        vaultConfig = new VaultConfig().address(vaultUrl.get());
         try {
-            LOGGER.log(Level.FINE, "Attempting to connect to Vault: {0}", vaultUrl.get());
+            LOGGER.log(Level.FINE, "Attempting to connect to Vault: {0}", vaultUrl);
             if (vaultNamespace.isPresent()) {
                 vaultConfig.nameSpace(vaultNamespace.get());
                 LOGGER.log(Level.FINE, "Using namespace with Vault: {0}", vaultNamespace);
             }
 
-            vaultConfig.engineVersion(Integer.parseInt(vaultEngineVersion.get()));
+            vaultConfig.engineVersion(Integer.parseInt(vaultEngineVersion));
             LOGGER.log(Level.FINE, "Using engine version: {0}", vaultEngineVersion);
 
             vaultConfig = vaultConfig.build();
         } catch (VaultException e) {
             LOGGER.log(Level.WARNING, "Could not configure vault connection", e);
         }
 
-        Vault vault = new Vault(vaultConfig);
-        Optional<String> authToken = Optional.empty();
-
-        // attempt token login
-        if (vaultToken.isPresent()) {
-            authToken = vaultToken;
-            LOGGER.log(Level.FINE, "Using supplied token to access Vault");
+        try {
+            vaultConfig.build();
```

Comment:
`vaultConfig.build();` called again in the same scope just another try block with a different error message :smile:

Reply:
Fixed :+1: